### PR TITLE
Switch the MVVM Toolkit to 8.2.0

### PR DESCRIPTION
**Pull request by Sergio0694**
## Summary of the pull request

This PR bumps the MVVM Toolkit to 8.2.0 and applies the necessary fixes for the new analyzers.

## Validation steps performed

Build the solution to ensure no warnings/errors, run and played with the app for a bit.

## PR checklist
- [X] Contributes to #841
- [ ] Tests added/passed (no tests changes)
- [ ] Documentation updated

### CODE REVIEW

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/SummaryViewModel.cs [hunk 1]
```diff
@@ -106,7 +106,7 @@ public ObservableCollection<PackageViewModel> AppsDownloaded
     [RelayCommand]
     public void RemoveRestartGrid()
     {
-        _showRestartNeeded = Visibility.Collapsed;
```
Comment (AmelBawa-msft) [inline tools/SetupFlow/DevHome.SetupFlow/ViewModels/SummaryViewModel.cs:109]: Bindings were never notified of this visibility update, was this working before the change?

**Review by AmelBawa-msft — APPROVED** [review]

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/SummaryViewModel.cs (hunk 1 above)
Comment (Sergio0694) [reply tools/SetupFlow/DevHome.SetupFlow/ViewModels/SummaryViewModel.cs:109]: I'm inclined to assume this was a bug before.

**Review by krschau — APPROVED** [review]

**Comment by Sergio0694** [comment]
Mmmh... Rebased after #1020 to trigger the CI, and it's failing with these errors:

```
Failed to download package 'CommunityToolkit.Mvvm.8.2.0' from 'https://pkgs.dev.azure.com/ms/fef2e43c-84a1-4265-b74c-b824d387dd06/_packaging/3989942b-0a74-47bc-a7e3-ba19148bd858/nuget/v3/flat2/communitytoolkit.mvvm/8.2.0/communitytoolkit.mvvm.8.2.0.nupkg'.
  Response status code does not indicate success: 401 (Unauthorized - No local versions of package 'communitytoolkit.mvvm'; please provide authentication to access versions from upstream that have not yet been saved to your feed. (DevOps Activity ID: 36D9F4A8-8AFF-4750-B7F8-8931369A8388)).
  Failed to download package 'CommunityToolkit.Mvvm.8.2.0' from 'https://pkgs.dev.azure.com/ms/fef2e43c-84a1-4265-b74c-b824d387dd06/_packaging/3989942b-0a74-47bc-a7e3-ba19148bd858/nuget/v3/flat2/communitytoolkit.mvvm/8.2.0/communitytoolkit.mvvm.8.2.0.nupkg'.
  Response status code does not indicate success: 401 (Unauthorized - No local versions of package 'communitytoolkit.mvvm'; please provide authentication to access versions from upstream that have not yet been saved to your feed. (DevOps Activity ID: 3EDAA527-45DE-4033-A8BB-AEA1D45E3E17)).
  Failed to download package 'CommunityToolkit.Mvvm.8.2.0' from 'https://pkgs.dev.azure.com/ms/fef2e43c-84a1-4265-b74c-b824d387dd06/_packaging/3989942b-0a74-47bc-a7e3-ba19148bd858/nuget/v3/flat2/communitytoolkit.mvvm/8.2.0/communitytoolkit.mvvm.8.2.0.nupkg'.
  Response status code does not indicate success: 401 (Unauthorized - No local versions of package 'communitytoolkit.mvvm'; please provide authentication to access versions from upstream that have not yet been saved to your feed. (DevOps Activity ID: E2B08F1B-C704-414A-9C60-A5CB7AB77EE9)).
  Failed to download package 'CommunityToolkit.Mvvm.8.2.0' from 'https://pkgs.dev.azure.com/ms/fef2e43c-84a1-4265-b74c-b824d387dd06/_packaging/3989942b-0a74-47bc-a7e3-ba19148bd858/nuget/v3/flat2/communitytoolkit.mvvm/8.2.0/communitytoolkit.mvvm.8.2.0.nupkg'.
  Response status code does not indicate success: 401 (Unauthorized - No local versions of package 'communitytoolkit.mvvm'; please provide authentication to access versions from upstream that have not yet been saved to your feed. (DevOps Activity ID: 885487FE-FF1C-4912-8889-802C5A4EAD87)).
  Failed to download package 'CommunityToolkit.Mvvm.8.2.0' from 'https://pkgs.dev.azure.com/ms/fef2e43c-84a1-4265-b74c-b824d387dd06/_packaging/3989942b-0a74-47bc-a7e3-ba19148bd858/nuget/v3/flat2/communitytoolkit.mvvm/8.2.0/communitytoolkit.mvvm.8.2.0.nupkg'.
  Response status code does not indicate success: 401 (Unauthorized - No local versions of package 'communitytoolkit.mvvm'; please provide authentication to access versions from upstream that have not yet been saved to your feed. (DevOps Activity ID: 9AC3A8A6-CF00-469A-88E6-0CC5646F7B1F)).
C:\Program Files\dotnet\sdk\7.0.302\NuGet.targets([15](https://github.com/microsoft/devhome/actions/runs/5164879456/jobs/9304047527#step:7:16)6,5): error : Failed to download package 'CommunityToolkit.Mvvm.8.2.0' from 'https://pkgs.dev.azure.com/ms/fef2e43c-84a1-4265-b74c-b824d387dd06/_packaging/3989942b-0a74-47bc-a7e3-ba19148bd858/nuget/v3/flat2/communitytoolkit.mvvm/8.2.0/communitytoolkit.mvvm.8.2.0.nupkg'. [D:\a\devhome\devhome\DevHome.sln]
C:\Program Files\dotnet\sdk\7.0.302\NuGet.targets(156,5): error : Response status code does not indicate success: 401 (Unauthorized - No local versions of package 'communitytoolkit.mvvm'; please provide authentication to access versions from upstream that have not yet been saved to your feed. (DevOps Activity ID: 88548279-FF1C-4912-8889-802C5A4EAD87)). [D:\a\devhome\devhome\DevHome.sln]
```

Do we need to update the upstream sources for the feed DevHome uses to build or something? 🤔
I thought it was already there since 8.0.0 was being used before this PR, so not entirely sure I get it.

**Comment by EricJohnson327** [comment]
@Sergio0694 Because of how upstream nuget feeds work in ADO, an admin needs to approve new packages.  Approved 8.2.0 and kicked off a rerun.

**Comment by Sergio0694** [comment]
Oh I see, didn't know you needed explicit approval on individual package versions (I don't think that's enabled by default).
Thank you for approving it and re-triggering the CI! 🙂

**Pull request merged**